### ISSUE TITLE
Consistent "formatting" for OffsetDateTime

### ISSUE THREAD

**Issue by reubenfirmin** [jdbi/jdbi#1507]
OffsetDateTime apparently has latitude in how it formats a date. I.e. "2018-01-01 00:00:00-05" retrieved from postgres and coerced to OffsetDateTime by jdbi could end up as OffsetDateTime("2018-01-01T00:00-05:00"), or OffsetDateTime("2018-01-01T05:00Z"), or OffsetDateTime("2018-01-01T01:00-04:00") depending on the machine timezone. These all represent the same instant, so technically there's no problem.

However, it makes writing tests that work on machines across different timezones a pain, because these different formats don't equal each other. To work around it, we wrote an OffsetDateTimeColumnMapper for our tests which coerces all reads to the UTC "format".

1) Request: JDBI should always return UTC formatted OffsetDateTimes, regardless of the machine local time, OR it should consistently return the identical zone to whatever surfaced from postgres. In the example result from postgres, it should return OffsetDateTime("2018-01-01 00:00:00-05") OR OffsetDateTime("2018-01-01T05:00Z), but in either case it should return the same across all machines (no matter their timezone.)

2) Suspected bug: my machine is in offset -4 right now. If I run the following code I get OffsetDateTimes with offsets -4:

```
        val instant = Instant.now() 
        val systemZone = ZoneId.systemDefault() // my timezone
        val currentOffsetForMyZone = systemZone.rules.getOffset(instant)
        return OffsetDateTime.parse(date, 
              DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm[:ss] Z"))
                .withOffsetSameInstant(currentOffsetForMyZone)
```

Postgres agrees:

```
SELECT EXTRACT(TIMEZONE FROM now())/3600.0;
 ?column? 
----------
       -4
(1 row)
```

However, on my machine jdbi is currently producing OffsetDateTime with an offset of -5 (i.e. apparently not realizing I'm in daylight savings.)

**Comment by qualidafial** [comment]
The `TIMESTAMP WITH TIME ZONE` data type in Postgres is misleading--any offset date time you supply to Postgres is converted to UTC and stored without any timezone information. When you fetch it from the database, it is converted to the "current" timezone--that is, your machine's time one.

https://www.postgresql.org/docs/9.1/datatype-datetime.html

> For timestamp with time zone, the internally stored value is always in UTC (Universal Coordinated Time, traditionally known as Greenwich Mean Time, GMT). An input value that has an explicit time zone specified is converted to UTC using the appropriate offset for that time zone. If no time zone is stated in the input string, then it is assumed to be in the time zone indicated by the system's timezone parameter, and is converted to UTC using the offset for the timezone zone.
> 
> When a timestamp with time zone value is output, it is always converted from UTC to the current timezone zone, and displayed as local time in that zone. To see the time in another time zone, either change timezone or use the AT TIME ZONE construct (see Section 9.9.3).

**Comment by reubenfirmin** [comment]
Yes, which is why I put quotes around "format". Don't you think the database access library should produce equal results for the same query and the same data on different machines, whereever they are? It's the same instant, i.e. the same fundamental data.

Also, please check out the suspected bug I described (#2).

**Comment by qualidafial** [comment]
Actually, I _don't_ think Jdbi should be inserting itself into this situation. The OffsetDateTime was returned from the Postgres database itself. It's not the library's job to interpret whether the database data fits your expectations--the data is the data.

If you want all times in UTC, I suggest modeling your column with an Instant instead of OffsetDateTime.

**Comment by qualidafial** [comment]
I will look into the bug you mentioned. What time zone is your machine set to?

**Comment by reubenfirmin** [comment]
OK.

My machine is in "America/New York".

**Comment by stevenschlansker** [comment]
I think the linked issue above is not correct?  It seems very old and unrelated.
